message.go

```diff
@@ -251,3 +287,200 @@ func extractOffset(base int64, msgSet []byte) (offset int64, err error) {
 	offset = base - offset
 	return
 }
+
+type messageSetHeaderV2 struct {
+	firstOffset          int64
+	length               int32
+	partitionLeaderEpoch int32
+	magic                int8
+	crc                  int32
+	batchAttributes      int16
+	lastOffsetDelta      int32
+	firstTimestamp       int64
+	maxTimestamp         int64
+	producerId           int64
+	producerEpoch        int16
+	firstSequence        int32
+}
+
+type compression int8
+
+const (
```

Comment:
I don't see these used anywhere.  Do we need them?  The compression codes should already be registered in `compression.go`